internal/integrations/opaque/opaque.go

```diff
@@ -85,12 +85,9 @@ func (OpaqueIntegration) PrepareRun(ctx context.Context, server planning.Server,
 			return err
 		}
 
-		config := binary.Config
-		if config != nil {
-			run.WorkingDir = config.WorkingDir
-			run.Command = config.Command
-			run.Args = config.Args
-			run.Env = config.Env
+		err = run.BinaryConfig.MergeFrom(binary.Config)
```

Comment:
Leaving our merge semantics to proto merge semantics feels like a future problem. Lets define our own merge function, with clear documentation of what gets replaced by what.

Reply:
(I see now that you've added these methods to the protos -- lets not do that, see my comment below regarding OOP).